pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,438 @@
+                dmap_data = dmap2dict(dmap_data)
+        except KeyError:
+            raise rtp_exceptions.RTPUnknownParameter(parameter)
+        cls.dmap_data = dmap_data
+        cls.__check_data_type(parameter, 'array')
+        # Calculate the time span range, either passed in or calculated
+        # based on the dmap_data records
+        # TODO: move this to it's own method
+        try:
+            if len(kwargs['time_span']) == 2:
+                start_time = kwargs['time_span'][0]
+                end_time = kwargs['time_span'][1]
+                cls.interval_time = None
+            elif len(kwargs['time_span']) == 3:
+                start_time = kwargs['time_span'][0]
+                end_time = kwargs['time_span'][2]
+                cls.interval_time = kwargs['time_span'][1]
+            else:
+                raise IndexError("time_span list must be length of 2 or 3")
+
+        except KeyError:
+            start_time = cls.__time2datetime(cls.dmap_data[0])
+            end_time = cls.__time2datetime(cls.dmap_data[-1])
+
+        # y-axis coordinates, i.e., range gates,
+        # TODO: implement variant other coordinate systems for the y-axis
+        y = np.linspace(0, cls.dmap_data[0]['nrang'],
+                        cls.dmap_data[0]['nrang'])
+        y_max = cls.dmap_data[0]['nrang']
+
+        # z: parameter data mapped into the color mesh
+        z = np.zeros((1, y_max)) * np.nan
+
+        # x: time date data
+        x = []
+
+        # We cannot simply use numpy's built in min and max function
+        # because of the ground scatter value
+        if settings["boundary"]:
+            z_min, z_max = settings["boundary"]
+        else:
+            z_min = cls.dmap_data[0][parameter][0]
+            z_max = cls.dmap_data[0][parameter][0]
+        for dmap_record in cls.dmap_data:
+            # get time difference to test if there is some gap data
+            time = cls.__time2datetime(dmap_record)
+            diff_time = 0.0
+            if time > end_time:
+                break
+            if x != []:
+                # 60.0 seconds in a minute
+                delta_diff_time = (time - x[-1])
+                diff_time = delta_diff_time.seconds/60.0
+
+            # separation roughly 2 minutes
+            if diff_time > 2.0:
+                # if there is gap data (no data recorded past 2 minutes)
+                # then fill it in with white space
+                for _ in range(0, int(np.floor(diff_time/2.0))):
+                    x.append(x[-1] + timedelta(0, 120))
+                    i = len(x) - 1  # offset since we start at 0 not 1
+                    if i > 0:
+                        z = np.insert(z, len(z), np.zeros(1, y_max) * np.nan,
+                                      axis=0)
+
+            # Get data for the provided beam number
+            if (beam_num == 'all' or dmap_record['bmnum'] == beam_num) and\
+               (settings['channel'] == 'all' or
```

Comment:
I believe you don't need the `\` for a line break within a parenthetical expression.